### PR TITLE
bpo-23216: IDLE: Add docstrings to search modules

### PR DESCRIPTION
**Pull request by csabella**
* Enhance docstrings for replace, grep, and search.
* Minor refactoring of boolean arguments to improve readability.

https://bugs.python.org/issue23216

### CODE REVIEW

File: Lib/idlelib/replace.py [hunk 1]
```diff
@@ -10,9 +10,16 @@
...
+
 def replace(text):
-    """Returns a singleton ReplaceDialog instance.The single dialog
-     saves user entries and preferences across instances."""
+    """Create or reuse a singleton ReplaceDialog instance.
+
+    The singleton dialog saves user entries and preferences
+    across instances.
+
+    Args:
```
Comment (eamanu) [inline Lib/idlelib/replace.py:20]: Why don't use ```Parameters```?

Comment (csabella) [reply Lib/idlelib/replace.py:20]: Mostly consistency.  Other modules in IDLE use `Args:`.  

There are different ways to do list the pararmeters and pep257 isn't specific about which one to use.  There is the java style, Sphinx/reST style, Google style, etc.

File: Lib/idlelib/replace.py [hunk 2]
```diff
@@ -180,12 +238,24 @@ def do_replace(self):
...
+        section is brought into view.
+
+        The colors from the 'hit' tag aren't currently shown
+        when the text is displayed.  This is due to the 'sel'
+        tag being added first, so the colors in the 'sel'
+        config are seen instead of the colors for 'hit'.
+
+        Args:
+            first: Starting text index.
+            last: Ending text index.
```
Comment (terryjreedy) [inline Lib/idlelib/replace.py]: These are self explanatory from first line, so this Args section is not needed.

File: Lib/idlelib/replace.py [hunk 3]
```diff
@@ -44,37 +71,54 @@ def open(self, text):
...
-        self.make_button("Replace+Find", self.default_command, 1)
+        self.make_button("Replace+Find", self.default_command, isdef=True)
         self.make_button("Replace All", self.replace_all)
 
     def find_it(self, event=None):
-        self.do_find(0)
+        """Handle the Find button.
+
+        In the Replace dialog, repeatedly using Find does not
+        go to the next occurrence of the pattern.
```
Comment (terryjreedy) [inline Lib/idlelib/replace.py]: It does for me.  If it does not on linux, it would be a bug to be fixed.

**Review by terryjreedy — CHANGES_REQUESTED** [review]
Address behavior discrepancy, remove 1 redundancy.
I appreciate the 0/1 to False/True changes.  They now look archaic and a bit jarring.

File: Lib/idlelib/replace.py (hunk 3 above)
Comment (csabella) [reply Lib/idlelib/replace.py]: This works for me on Windows and Linux now.  I'm not sure why I thought it didn't.  If I change the call to `self.do_find(True)`, then it won't go to the next occurrence, so maybe I had the flag set wrong when I was testing it.  Sorry about the false alarm.

File: Lib/idlelib/replace.py (hunk 2 above)
Comment (csabella) [reply Lib/idlelib/replace.py]: Removed.

**Comment by csabella** [comment]
I have made the requested changes; please review again.

**Review by terryjreedy — APPROVED** [review]

**Comment by terryjreedy** [comment]
False positive.  All tests passed but there was a failure in the profiling or something like that, which I have not seen before.

**Pull request merged**